contracts/dapis/OevDataFeedServer.sol

```diff
@@ -50,10 +50,26 @@ contract OevDataFeedServer is DataFeedServer, IOevDataFeedServer {
         bytes calldata data,
         bytes[] calldata packedOevUpdateSignatures
     ) external payable override onlyValidTimestamp(timestamp) {
-        require(
-            timestamp > _oevProxyToIdToDataFeed[oevProxy][dataFeedId].timestamp,
-            "Does not update timestamp"
-        );
+        int224 updatedValue = decodeFulfillmentData(data);
+        {
+            DataFeed storage oevProxyDataFeed = _oevProxyToIdToDataFeed[
+                oevProxy
+            ][dataFeedId];
+            require(
+                timestamp > oevProxyDataFeed.timestamp,
+                "Does not update timestamp"
+            );
+            if (oevProxyDataFeed.timestamp != 0) {
+                require(
+                    updatedValue != oevProxyDataFeed.value,
+                    "Does not update value"
+                );
+            }
+            require(
+                updatedValue != _dataFeeds[dataFeedId].value,
+                "Updated value same as base feed"
+            );
```

Comment:
> Considering that an OEV update with the same value as the base data feed doesn't make much sense

I have to be honest and say that I don't quite understand why this is true. If base value of ETH/USD is 2000 and OEV value is 1900 then if we get an OEV update for 2000 shouldn't that update the OEV value regardless of being the same as the base value?

Reply:
I think you're right. Say the current state is this
- base feed (value: 2000, timestamp: 100)
- oev feed (value: 1900, timestamp: 90)

When this is read by the OEV proxy, it will already return (value: 2000, timestamp: 100). Then, updating the OEV value to 2000 has no use other than updating the timestamp to something larger than 100 while keeping the value the same, which is what I was trying to address here.

Alternatively, say the current state is
- base feed (value: 2000, timestamp: 100)
- oev feed (value: 1900, timestamp: 110)

When this is read by the OEV proxy, it will return (1900, 110). Here, updating the OEV value to 2000 is legitimate because the OEV update at timestamp: 110 is proof that the data feed wasn't flatlined since timestamp: 100, so we're not updating a flatline. Preventing this legitimate update is similar to how the APIs for a data feed can return the exact same value after a heartbeat interval and cause the legitimate heartbeat to fail. Our assumption here is that the off-chain sources for a live data feed change all the time, so this will at most cause the heartbeat to be delayed by a couple of minutes and the consequences of a data feed going stale is much milder than a misreport, it will just cause the user's project to not allow interactions until the update.

The problem here is similar in that even if the OEV value can't be updated to 2000, it will soon be updated to 1999 or 2001. Nevertheless, replacing [this](https://github.com/api3dao/airnode-protocol-v1/blob/1d173b8ded7bffe5f32dd0a60b225fbf6b6df103/contracts/dapis/OevDataFeedServer.sol#L68-L71) with
```solidity
DataFeed storage baseDataFeed = _dataFeeds[dataFeedId];
if (updatedValue == baseDataFeed.value) {
  require(oevProxyDataFeed.timestamp > baseDataFeed.timestamp, "Data source may have flatlined"); // revert string suggestions welcome
}
```
is probably prudent (and the added gas cost is not a problem because this is not the happy path).